plugins/generic/usageStats/UsageStatsPlugin.inc.php

```diff
@@ -272,7 +306,49 @@ function getUsageEventCurrentDayLogName() {
 	 * @param $usageEvent array
 	 */
 	function _writeUsageEventInLogFile($usageEvent) {
-		$desiredParams = array($usageEvent['ip']);
+		import('lib.pkp.classes.file.PrivateFileManager');
+		$fileMgr = new PrivateFileManager();
+
+		// Check the plugin file directory.
+		$usageEventFilesPath = $this->getUsageEventLogsPath();
+		if (!$fileMgr->fileExists($usageEventFilesPath, 'dir')) {
+			$success = $fileMgr->mkdirtree($usageEventFilesPath);
+			if (!$success) {
+				// Files directory wrong configuration?
+				assert(false);
+				return false;
+			}
+		}
+		// Get the current day filename.
+		$filename = $this->getUsageEventCurrentDayLogName();
+		$filePath = $usageEventFilesPath . DIRECTORY_SEPARATOR . $filename;
+
+		// Salt management.
+		$salt = $this->getSetting(0, 'salt');
```

Comment:
I think I am -- because of this condition: !file_exists($filePath) -- the filePath is changing each day -- it will ge benerated each day. Correct?